docs/5.x/writing-a-record-builder.md

```diff
@@ -0,0 +1,227 @@
+with a variable number of rows, it's good practice to make sure the number of rows is capped.
+
+To set a limit, set the `maxRowsInTable` and `maxRowsInSubtable` properties in the constructor of your `RecordBuilder`.
+This can be hard-coded or it can come from configuration:
+
+```
+class MyRecordBuilder extends RecordBuilder
+{
+    public function __construct()
+    {
+        parent::__construct();
+        $this->maxRowsInTable = (int)Config::getInstance()->MyPlugin['datatable_archiving_maximum_rows'];
+        $this->maxRowsInSubtable = (int)Config::getInstance()->MyPlugin['datatable_archiving_maximum_rows_subtable'];
+
+        // we want to sort by the most important metric in our reports before we cut off rows
+        $this->columnToSortByBeforeTruncation = Metrics::INDEX_NB_VISITS;
+    }
+}
+```
+
+If you don't know what to use, you can set both values to `Config::getInstance()->General['datatable_archiving_maximum_rows_standard']`.
+
+Also note we set `columnToSortByBeforeTruncation` to make sure the rows with the least visits are the ones that get removed.
+
+Additionally, if your plugin provides metrics that should be aggregated together with an operation other than being `sum`-ed,
+you will need to set the `$columnAggregationOps` property:
+
+```
+class MyRecordBuilder extends RecordBuilder
+{
+    public function __construct()
+    {
+        parent::__construct();
+        
+        // ...
+
+        $this->columnAggregationOps = [
+            'my_max_metric' => 'max',
+            'my_min_metric' => 'min',
+            'my_other_metric' => function ($thisValue, $otherValue) {
+                // custom aggregation logic here
+            },
+        ];
+    }
+}
+```
+
+Note that each of these settings can also be overridden for specific records by setting the relevant property
+on `Record` instances in your `getRecordMetadata()` method.
+
+### Step five: if your RecordBuilder is parameterized, implement the relevant event
+
+If your `RecordBuilder` is not parameterized then there's nothing else to do. You're done and Matomo will detect and use it.
+
+If it is parameterized, then there's still one thing left to do. Matomo will not be able to automatically create a `RecordBuilder`
+that takes parameters, so it must be added manually in the `Archiver.addRecordBuilders` event like so:
+
+```
+class MyPlugin
+{
+    public function registerEvents()
+    {
+        $hooks = [
+            'Archiver.addRecordBuilders' => 'addRecordBuilders',
+        ];
+        return $hooks;
+    }
+
+    public function addRecordBuilders(array &$recordBuilders): void
+    {
+        $idSite = \Piwik\Request::fromRequest()->getIntegerParameter('idSite', 0);
+        if (!$idSite) {
+            return;
+        }
+
+        $entities = StaticContainer::get(MyEntityDao::class)->getAllEntitiesForSite($idSite);
+        foreach ($entities as $entity) {
+            $recordBuilders[] = new MyRecordBuilder($entity);
+        }
+    }
+}
```

Comment:
@diosmosis Should we also mention about `buildForNonDayPeriod` here, so that its easy for others to edit when required?

Reply:
I can try and add something when i have time

Reply:
added a quick section